4.student-exercises.js

```diff
@@ -0,0 +1,68 @@
+const getTrainersAndGymsAndPokemons = (gyms, trainers, pokemons) => {
+  // Combine trainers and pokemons and gyms
+  return (
+    trainers
+      // Map trainers to include their pokemons
+      .map(trainer => {
+        // Include the trainer's pokemons
+        trainer.pokemons = pokemons.filter(pokemon =>
+          trainer.pokemonIds.includes(pokemon.id)
+        );
+        // Include the trainer's gym
+        trainer.gym = gyms.find(gym => gym.trainerId === trainer.id);
+        return trainer;
```

Comment:
And then return an entirely new object with new values using the spread operator:
```suggestion
        return {
           ...trainer,
           pokemons: trainerPokemons,
           gym: trainerGym
        }
```